src/input/InputManager.cpp

```diff
@@ -40,13 +40,6 @@ CInputManager& CInputManager::Get(void)
   return _instance;
 }
 
-CInputManager::CInputManager()
-{
-  game_controller mouseControllerStruct = {};
-  mouseControllerStruct.controller_id = "game.controller.mouse";
-  m_mouseDevice = std::make_shared<CLibretroDevice>(&mouseControllerStruct);
-}
```

Comment:
you're right. `"game.controller.mouse"` definitely needs to come from the frontend though, so I see two choices: We either make UpdatePort() in the Game API accept a int, or we add a UpdateMouse() function. I prefer the former, because we've already defined a value for the mouse port.

An interesting side effect of the former option is that we can move all the keyboard handling out of CInputManager and into CLibretroDeviceInput.